client/src/index.js

```diff
@@ -1,12 +1,19 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import './index.css';
-import App from './App';
-import * as serviceWorker from './serviceWorker';
+import React from "react";
+import ReactDOM from "react-dom";
+import "./index.css";
+import App from "./App";
+import * as serviceWorker from "./serviceWorker";
+import { UserContextProvider } from "./provider/UserContext";
 
-ReactDOM.render(<App />, document.getElementById('root'));
+ReactDOM.render(
+  
+    <UserContextProvider>
```

Comment:
Can you wrap the inside of App.js with this Context Provider instead?

Reply:
i was not able to use `dispatch` in `App.js` by doing that. so i moved Context provider to `index.js`. i can try another way if something else works

Reply:
That's interesting, it should be functionally the same given how it's organized right now? Try it a little more with some console.logs and if not don't worry about it.